plugins/modules/vmware_vm_storage_policy.py

```diff
@@ -49,7 +49,7 @@
     - This parameter is ignored when C(state=absent).
     - This parameter is required when C(state=present).
     required: False
-    type: str
+    type: list
```

Comment:
This change will break existing playbooks. I would suggest adding a new parameter that will take `key:value` pair as a list where the key will be a category and the value will be a tag name something like this - 

```yaml
tag_info:
- tag_name: gold
   category_name: performance_tier
- tag_name: platinum
   category_name: cost_effective_tier
```

Make this new parameter mutually_exclusive with `tag_category` and `tag_name`.